src/valuations/OracleValuation.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity 0.8.28;
+
+import {IOracleValuation} from "./interfaces/IOracleValuation.sol";
+
+import {D18} from "../misc/types/D18.sol";
+
+import {PoolId} from "../common/types/PoolId.sol";
+import {AssetId} from "../common/types/AssetId.sol";
+import {PricingLib} from "../common/libraries/PricingLib.sol";
+import {ShareClassId} from "../common/types/ShareClassId.sol";
+import {IValuation} from "../common/interfaces/IValuation.sol";
+
+import {IHub} from "../hub/interfaces/IHub.sol";
+import {IHubRegistry} from "../hub/interfaces/IHubRegistry.sol";
+
+/// @notice Provides an implementation for valuation of assets by trusted price feeders.
+///         Prices should be denominated in the pool currency.
+///         Quorum is always 1, i.e. there is no aggregation of prices across multiple feeders.
+/// @dev    To set up, add a price feeder using hub.updateFeeder(), set this contract as the valuation
+///         for one or more assets, and set this contract as a hub manager, so it can call
+///         hub.updateHoldingValue().
+contract OracleValuation is IOracleValuation {
+    IHub public immutable hub;
+    IHubRegistry public immutable hubRegistry;
+
+    mapping(PoolId => mapping(address => bool)) public feeder;
+    mapping(PoolId => mapping(ShareClassId => mapping(AssetId base => Price))) public price;
```

Comment:
Nit question: Could we rename this to `pricePoolPerAsset` to be more explicit? Without the `OracleValuation` contract documentation, its not fully obvious IMO.

Reply:
Definitely!